graphql/database/users/deleteUser.js

```diff
@@ -0,0 +1,48 @@
+import uuid from 'uuid/v4'
+import UserModel from './UserModel'
+import UserWidgetModel from '../widgets/userWidget/UserWidgetModel'
+import getWidgets from '../widgets/getWidgets'
+
+/**
+ * Delete a user and performs other cleanup actions.
+ * This function must be idempotent, as the client may call it
+ * for existing users.
+ * @param {object} userContext - The user authorizer object.
+ * @param {string} userId - The user's ID.
+ */
+const deleteUser = async (userContext, userId) => {
+  try {
```

Comment:
Before proceeding, ensure the user has re-authenticated recently using the `auth_time` in claims. See ticket for details.